api/callbacks.go

```diff
@@ -365,9 +365,9 @@ var querier_vtable = C.Querier_vtable{
 
 // contract: original pointer/struct referenced must live longer than C.GoQuerier struct
 // since this is only used internally, we can verify the code that this is the case
-func buildQuerier(q Querier) C.GoQuerier {
+func buildQuerier(q *Querier) C.GoQuerier {
```

Comment:
Same point with interfaces.
In general, you never want to take pointers to interfaces, that is very odd behavior.
We do this as interfaces are 16-byte wide and we need to pass an 8-byte wide pointer over FFI.

Reply:
resolving following your new comment